internal/engine/wazevo/backend/machine.go

```diff
@@ -61,6 +61,12 @@ type (
 		// ResolveRelocations resolves the relocations after emitting machine code.
 		ResolveRelocations(refToBinaryOffset map[ssa.FuncRef]int, binary []byte, relocations []RelocationInfo)
 
+		// UpdateRelocationInfo recomputes the RelocationInfo after emitting machine code and returns the updated
+		// RelocationInfo and the size of the trampoline if necessary.
+		UpdateRelocationInfo(refToBinaryOffset map[ssa.FuncRef]int, trampolineOffset int, r RelocationInfo) (res RelocationInfo, trampolineSize int)
+
+		RelocationTrampolineSize(relocations []RelocationInfo) int
```

Comment:
Not really in love with this, but I could not think of anything else. We need to know the size estimate of a given trampoline "section" at the end of a function.

note to self: need doc comment